chain/sync.go

```diff
@@ -886,6 +886,35 @@ func (syncer *Syncer) syncFork(ctx context.Context, incoming *types.TipSet, know
 		}
 	}
 
+	incomingParentsTsk := incoming.Parents()
+	commonParent := false
+	for _, incomingParent := range incomingParentsTsk.Cids() {
+		if known.Contains(incomingParent) {
+			commonParent = true
+		}
+	}
+
+	if commonParent {
+		// known contains at least one of incoming's Parents => the common ancestor is known's Parents (incoming's Grandparents)
```

Comment:
Pretty sure this is a typo, as I parse it we only have the guarantee about common grandparents

![2023-12-15-223158_1222x936_scrot](https://github.com/filecoin-project/lotus/assets/3867941/1d39d256-027c-4468-ac52-7ded7d8afc3a)


```suggestion
		// known contains at least one of incoming's Parents => the common ancestor is known's Grandparents (also incoming's Grandparents)
```

Reply:
(the logic does appear to be correct tho)

Reply:
@magik6k No, I don't think this is right? The logic in 891-895 is checking whether `incoming.Parents()` has any blocks in common with `known` (so `known = {A,B,C,D}` in your diagram).